### PR TITLE
Enable Ladybug

### PR DESCRIPTION
**Pull request by jacodg**
Requires

- [x] SecurityFilterChainConfigurer needs to be fixed
- [ ] #6900
- [ ] #6901

### CODE REVIEW

**Comment by tnleeuw** [comment]
There are errors starting the Ladybug, see: https://jenkins.frankframework.org/job/ff-ci/job/larva/36528/

From the logs:

```
2024-05-27 10:39:27,286 INFO  [main] {} APPLICATION - Successfully started Frank EnvironmentContext
2024-05-27 10:39:27,286 DEBUG [main] {} APPLICATION - Starting Ladybug TestTool
2024-05-27 10:39:27,312 INFO  [main] {} APPLICATION - loading TestTool configuration [classpath:springIbisTestTool.xml]
2024-05-27 10:39:28,237 DEBUG [main] {} servlets.ServletConfiguration - converted raw mapping [/iaf/ladybug/api/*] to [/iaf/ladybug/api/*]
2024-05-27 10:39:28,237 INFO  [main] {} servlets.NoOpAuthenticator - registering private endpoint with url pattern [/iaf/ladybug/api/*]
2024-05-27 10:39:28,238 DEBUG [main] {} servlets.ServletConfiguration - converted raw mapping [/iaf/ladybug/*] to [/iaf/ladybug/*]
2024-05-27 10:39:28,238 INFO  [main] {} servlets.NoOpAuthenticator - registering private endpoint with url pattern [/iaf/ladybug/*]
2024-05-27 10:39:28,239 DEBUG [main] {} servlets.ServletConfiguration - converted raw mapping [/iaf/testtool] to [/iaf/testtool]
2024-05-27 10:39:28,239 INFO  [main] {} servlets.NoOpAuthenticator - registering private endpoint with url pattern [/iaf/testtool]
2024-05-27 10:39:28,299 FATAL [main] {} APPLICATION - Unable to start Ladybug TestTool
org.springframework.context.ApplicationContextException: Unable to start web server
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.onRefresh(ServletWebServerApplicationContext.java:165) ~[spring-boot-3.3.0.jar:3.3.0]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:618) ~[spring-context-6.1.8.jar:6.1.8]

...
Caused by: java.lang.IllegalStateException: Failed to register 'filter springSecurityFilterChain' on the servlet context. Possibly already registered?
	at org.springframework.boot.web.servlet.DynamicRegistrationBean.register(DynamicRegistrationBean.java:122) ~[spring-boot-3.3.0.jar:3.3.0]
	at org.springframework.boot.web.servlet.RegistrationBean.onStartup(RegistrationBean.java:52) ~[spring-boot-3.3.0.jar:3.3.0]

```

**Comment by jacodg** [comment]
Sorry, when creating the PR I overlooked that I had locally disabled SecurityFilterChainConfigurer to prevent this error while looking into other errors. I hope Niels can fix this error

**Review by tnleeuw — CHANGES_REQUESTED** [review]
Ladybug doesn't start, blocking framework startup. Needs to be fixed before this can be merged.

File: ladybug/src/main/java/org/frankframework/ibistesttool/tibet2/Storage.java [hunk 1]
```diff
@@ -223,7 +223,8 @@ public List<List<Object>> getMetadata(int maxNumberOfRecords,
 				query.append(metadataName);
 			}
 		}
-		String rowNumber = dbmsSupport.getRowNumber(metadataNames.get(0), "desc");
+//		String rowNumber = dbmsSupport.getRowNumber(metadataNames.get(0), "desc");
```
Comment (nielsm5) [inline ladybug/src/main/java/org/frankframework/ibistesttool/tibet2/Storage.java:226]: I've created issue #6900 in order to fix this!

File: ladybug/src/main/resources/springIbisTestTool.xml [hunk 2]
```diff
@@ -77,7 +87,7 @@
 	</task:scheduled-tasks>
 
 	<bean name="dataSourceName" class="java.lang.String">
-		<constructor-arg value="${ladybug.jdbc.datasource:${jdbc.datasource.default}}"/>
+		<!-- constructor-arg value="${ladybug.jdbc.datasource:${jdbc.datasource.default}}"/ -->
```
Comment (nielsm5) [inline ladybug/src/main/resources/springIbisTestTool.xml:90]: I've created issue #6901 to re-enable this.

**Review by nielsm5 — APPROVED** [review]

File: ladybug/src/main/java/org/frankframework/ibistesttool/jdbc/LadyBugDataSourceFactory.java [hunk 3]
```diff
@@ -77,44 +77,44 @@ private static boolean isPooledDataSource(DataSource dataSource) {
 	}
 
 	private DataSource createPool(DataSource dataSource, String dataSourceName) {
-		if (isPooledDataSource(dataSource)) {
+//		if (isPooledDataSource(dataSource)) {
```
Comment (tnleeuw) [inline ladybug/src/main/java/org/frankframework/ibistesttool/jdbc/LadyBugDataSourceFactory.java:80]: Zullen we alle uit-gecommentariseerde code er maar gelijk uithalen?

File: ladybug/src/main/resources/springIbisTestTool.xml (hunk 2 above)
Comment (tnleeuw) [inline ladybug/src/main/resources/springIbisTestTool.xml:90]: Datasource name moet leeg blijven?

Comment (nielsm5) [reply ladybug/src/main/resources/springIbisTestTool.xml:90]: [Ja zie mijn bovenstaande comment](https://github.com/frankframework/frankframework/pull/6876#discussion_r1618628457)

**Review by tnleeuw — APPROVED** [review]

**Pull request merged**